### PR TITLE
feat: Not  expr to string

### PR DESCRIPTION
**Pull request by sebastian2296**
## Which issue does this PR close?

This PR adds support for _Not_ expr in #9726

### CODE REVIEW

**Comment by alamb** [comment]
Thanks @sebastian2296 -- if CI passes this PR looks great to me

**Review by alamb — APPROVED** [review]

File: datafusion/sql/src/unparser/expr.rs [hunk 1]
```diff
@@ -711,6 +718,10 @@ mod tests {
                 Expr::IsUnknown(Box::new((col("a") + col("b")).gt(lit(4)))),
                 r#"(("a" + "b") > 4) IS UNKNOWN"#,
             ),
+            (
+                Expr::Not(Box::new(col("a"))),
```
Comment (alamb) [inline datafusion/sql/src/unparser/expr.rs:722]: I think you could write this more concisely using https://docs.rs/datafusion/latest/datafusion/logical_expr/fn.not.html

So something like `not(col(a))`

**Comment by sebastian2296** [comment]
Hi @alamb, clippy test is failing due to unused ´not´ import, but that's not true. I'm using it to perform the test. Any hints on what could be the issue here?

**Comment by devinjdangelo** [comment]
> Hi @alamb, clippy test is failing due to unused ´not´ import, but that's not true. I'm using it to perform the test. Any hints on what could be the issue here?

Hey @sebastian2296 if you are using an import only for a test, you can import it within the mod tests {} block. E.g.

```rust
use regular_dependency;

fn some_code(){
  ()
}

mod tests{
    use test_dependency;
    ...

}
```

Thank you for the work on this PR! :pray:

**Comment by alamb** [comment]
🚀  -- thanks again @sebastian2296

**Comment by alamb** [comment]
Thanks @sebastian2296  -- this PR needs to be merged up from main to resolve a conflct. Can you do so so that I can merge it in?

Thanks again 🙏

**Comment by sebastian2296** [comment]
Done! @alamb

**Comment by alamb** [comment]
Thanks @sebastian2296  -- a very nice addition indeed

**Pull request merged**